contrib/utilities/prm_tools.py

```diff
@@ -0,0 +1,315 @@
+                file.write("\n")
+            write_comment_if_existent(parameters[entry]["comment"], prepend_spaces, file)
+
+            # Write the include
+            file.write(" " * prepend_spaces + "include " + entry + "\n")
+            # Add an empty line after include, except for the end of the file
+            if entry != list(parameters)[-1]:
+                file.write("\n")
+
+        elif parameters[entry]["type"] == "subsection":
+            if first_entry == False:
+                file.write("\n")
+            write_comment_if_existent(parameters[entry]["comment"], prepend_spaces, file)
+
+            file.write(" " * prepend_spaces + "subsection " + entry + "\n")
+            prepend_spaces += 2
+            # Recursively write the next subsection level
+            write_value_or_subsection(parameters[entry]["value"], prepend_spaces, file)
+            prepend_spaces -= 2
+            file.write(" " * prepend_spaces + "end\n")
+
+        elif parameters[entry]["type"] == "comment":
+            if first_entry == False and parameters[entry]["comment"] != "":
+                file.write("\n")
+            write_comment_if_existent(parameters[entry]["comment"], prepend_spaces, file)
+
+        first_entry = False
+
+    return 0
+
+
+
+def write_parameter_file(parameters, file):
+    """ Given a dictionary of parameters with a structure as
+    the one created by read_parameter_file(), write the dictionary
+    to a file.
+    """
+
+    prepend_spaces = 0
+    with open(file,'w') as f:
+        write_value_or_subsection(parameters, prepend_spaces, f)
+    return 0
+
+
+
+def main(input_file, output_file):
+    """ This function reformats the given input .prm files to follow our
+    general formatting guidelines. These are:
+
+    - indent by two spaces for each subsection level, and indent comments as well
+    as content
+    - ensure an empty line before a subsection or a comment, unless it is a
+    subsection/comment following directly another subsection, or it is at the
+    start of the file
+    - combine content of subsections that are duplicated
+    - if values are duplicated, merge them and use the last one in the file (like
+    deal.II ParameterHandler when it reads the file)
+    - retain as much user formatting in comments and parameter values as possible,
+    e.g. spaces for padding values to align between adjacent lines. This is not
+    always perfectly possible.
+    - retain broken lines (`\\`) in values of parameters and comments, remove them
+    from subsection or parameter names.
+    """
+    parameters = read_parameter_file(input_file)
+    write_parameter_file(parameters, output_file)
+
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser(
+                    prog='deal.II .prm file reformatter',
+                    description='Reformats deal.II .prm files to follow our general formatting guidelines. See the documentation of this script for details.')
```

Comment:
I think it would be nice to print the intended use (i.e., the list of arguments) here.